syntax = "proto3";

package tinykvpp.v1;

service TinyKVPPService {
  rpc Put(PutRequest) returns (PutResponse);
  rpc Get(GetRequest) returns (GetResponse);
}

message PutRequest {
  string key = 1;
  string value = 2;
}

message PutResponse {
  bool success = 1;
  string error = 2;
}

message GetRequest {
  string key = 1;
}

message GetResponse {
  bool found = 1;
  string value = 2;
}

// For internal database use
message DatabaseOperation {
  enum Type {
    TYPE_UNSPECIFIED = 0;
    TYPE_PUT = 1;
    TYPE_DELETE = 2;
    TYPE_BATCH = 3;
  }

  Type type = 1;
  string key = 2;
  string value = 3;
  // Reserve 4 for Batch operations
  uint64 request_id = 5;
}
